api/objectstore.go

```diff
@@ -0,0 +1,193 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/util"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher/api"
+	"github.com/rancher/go-rancher/client"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (s *Server) ObjectStoreList(rw http.ResponseWriter, req *http.Request) (err error) {
+	apiContext := api.GetApiContext(req)
+	col, err := s.objectStoreList(apiContext)
+	if err != nil {
+		return err
+	}
+	apiContext.Write(col)
+	return nil
+}
+
+func (s *Server) objectStoreList(apiContext *api.ApiContext) (*client.GenericCollection, error) {
+	list, err := s.m.ListObjectStoresSorted()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list object stores")
+	}
+
+	data := []interface{}{}
+
+	for _, store := range list {
+		vol, err := s.m.Get(fmt.Sprintf("pv-%v", store.Name))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get volume for %v", store.Name)
+		}
+		data = append(data, toObjectStoreResource(store, vol.Spec.Size, vol.Status.ActualSize))
+	}
+
+	return &client.GenericCollection{
+		Data: data,
+		Collection: client.Collection{
+			ResourceType: "objectStore",
+		},
+	}, nil
+}
+
+func (s *Server) ObjectStoreGet(rw http.ResponseWriter, req *http.Request) (err error) {
+	apiContext := api.GetApiContext(req)
+	resp := &client.GenericCollection{}
+	apiContext.Write(resp)
+	return nil
+}
```

Comment:
This function has not implemented

Reply:
I start to doubt that we need it since the UI only lists object stores but doesn't have a detailed view like the volumes have.
I'll double check and remove it if possible.